jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -83,6 +83,9 @@ public WebSocketClient()
      */
     public WebSocketClient(HttpClient httpClient)
     {
+        ByteBufferPool bufferPool = httpClient != null ? httpClient.getByteBufferPool() : null;
```

Comment:
This implementation now respects this constructor's javadoc about using the `HttpClient` _to base internal defaults off of_.

But I think you should add an extra constructor that takes both a `HttpClient` as well as a `WebSocketComponents` as the latter contains other things one might want to customize.